pkg/controller/migplan/registry_health_failed.go

```diff
@@ -0,0 +1,129 @@
+package migplan
+
+import (
+	"github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"time"
+)
+
+// A registry health checker which routinely checks whether registries are healthy or not for migplan's failed migrations
+type registryHealthFailed struct {
```

Comment:
looks like everything for this and `registryHealthRunning` is the same except the switch cases which check for the conditions, can you take in a function pointer to check for cases and reuse the other common code?

Reply:
Combined the watches.

Reply:
+1 looks great, thanks